### PR TITLE
Upgrade to badger V3

### PR DESCRIPTION
**Pull request by koko1123**
## Goal of this PR
Switch badger to V3 from https://github.com/outcaste-io/badger/tags

- tests pass

### CODE REVIEW

File: service/index/reader.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 
 	"golang.org/x/sync/errgroup"
 
-	"github.com/dgraph-io/badger/v2"
+	"github.com/dgraph-io/badger/v3"
```
Comment (diaswrd) [inline service/index/reader.go:24]: This is the one we're looking for 🔥

**Review by diaswrd — APPROVED** [review]

**Review by SaveTheRbtz — APPROVED** [review]

File: service/metrics/badger.go [hunk 2]
```diff
@@ -18,25 +18,25 @@ import (
...
-		"badger_v2_puts_total":           prometheus.NewDesc("archive_badger_puts_total", "number of puts", nil, nil),
-		"badger_v2_blocked_puts_total":   prometheus.NewDesc("archive_badger_blocked_puts_total", "number of blocked puts", nil, nil),
-		"badger_v2_pending_writes_total": prometheus.NewDesc("archive_badger_badger_pending_writes_total", "tracks the number of pending writes", []string{"path"}, nil),
-		"badger_v2_lsm_bloom_hits_total": prometheus.NewDesc("archive_badger_lsm_bloom_hits_total", "number of LSM bloom hits", []string{"level"}, nil),
-		"badger_v2_lsm_level_gets_total": prometheus.NewDesc("archive_badger_lsm_level_gets_total", "number of LSM gets", []string{"level"}, nil),
-		"badger_v2_lsm_size_bytes":       prometheus.NewDesc("archive_badger_lsm_size_bytes", "size of the LSM in bytes", []string{"path"}, nil),
-		"badger_v2_vlog_size_bytes":      prometheus.NewDesc("archive_badger_vlog_size_bytes", "size of the value log in bytes", []string{"path"}, nil),
+		"badger_v4_disk_reads_total":     prometheus.NewDesc("archive_badger_disk_reads_total", "cumulative number of reads", nil, nil),
```
Comment (SaveTheRbtz) [inline service/metrics/badger.go:27]: nit: v3

**Comment by SaveTheRbtz** [comment]
Before committing to master you may want to try deploying the package from this branch to verify that this fixes resource consumption issue

File: models/archive/badger.go [hunk 3]
```diff
@@ -15,24 +15,13 @@
...
-		WithValueLogLoadingMode(options.FileIO).
 		WithNumMemtables(1).
-		WithKeepL0InMemory(false).
 		WithCompactL0OnClose(false).
-		WithNumLevelZeroTables(1).
-		WithNumLevelZeroTablesStall(2).
-		WithLoadBloomsOnOpen(false).
-		WithIndexCacheSize(2000 << 20).
-		WithBlockCacheSize(0).
-		WithLogger(nil)
```
Comment (SaveTheRbtz) [inline models/archive/badger.go:37]: do you still need these changes or were they related to things cut in v4?

**Comment by koko1123** [comment]
parallel effort from @SaveTheRbtz closing.

**Pull request closed without merging**